### DISCUSSION TITLE
Subscription concurrency

### DISCUSSION THREAD

**Discussion by dedels** [ravendb/ravendb#15218] [General]
Our application needs to react to changes to a member based on incoming data.  We have a subscription attached to a map-reduce that triggers every time we get new data for the member.

However, if we are using concurrent subscriptions, if new data comes in during subscription processing, it can cause the save of data to fail.  

What should we do here?  
Should we allow this subscription item to fail?  It will be retried?
Since a newer event will trigger for this same member, we actually are ok if we lose processing on the first item.  Will this happen?

**Comment by georgiosd** [comment]
AFAIK the subscription will be triggered every time a matching document is changed, so it should be triggered again after the concurrent save.

But why not handle the concurrency error?

**Comment by ayende** [comment]
The way to handle that is to simply handle the concurrency exception, no?
Or just write _without_ checking for concurrency, and know that the value will be updated the next batch.

Are you writing to the same entries that you are subscribing on? How do you prevent a loop?
